### ISSUE TITLE
Root/parent/ancestor refs in nested structure

### ISSUE THREAD

**Issue by seanpan** [marchah/json-mapper-json#3]
Love this lib. Hope a useful feature can be supported: root/parent/ancestor refs in nested declaration.

**Comment by marchah** [comment]
@seanpan Thank you

can you give me an example please

**Comment by seanpan** [comment]
Thaks for your quick response. I was going to parse a json yesterday as below:

##### Raw JSON

```
{
  "content": {
    "result": [
      {
        "courseStatisticsDto": {
          "times": 3,
          "persons": 1,
          "courseCode": ""
        },
        "courseAddressDto": {},
        "endDate": 1460590552000,
        "startDate": 1460590552000,
        "name": "Example Course"
      }
    ],
    "type": "offline"
  }
}
```

##### Target JSON

```
{
  "data": [
    {
      "name": "Example Course",
      "code": "",
      "type": "offline"
    }
  ]
}
```

So I wrote a template:

##### Template

```
{
    data: {
        path: 'content.result',
        nested: {
            name: 'name',
            code: 'courseStatisticsDto.courseCode',
            //access point to the data root
            //type: '$root.content.type' 
        }
    }
}
```

However I found no way to manage the **type** property since it's outside the main array. It seems a root/parent/ancestor reference feature is required to support this scenario as I commented above.

Let me know your concern and if I misunderstand the usage.

**Comment by marchah** [comment]
@seanpan good idea

let me know what do you think and if it's like you wanted i will merge it:

https://github.com/marchah/json-mapper-json/tree/root-key-word-%233

**Comment by seanpan** [comment]
Exactly what I want. Thank for your brilliant work!

**Comment by marchah** [comment]
@seanpan thanks, available with npm on version `0.9.2`
